libcloud/utils/misc.py

```diff
@@ -313,29 +324,51 @@ def retry(retry_exceptions=EXCEPTION_TYPES, retry_delay=None,
     retry_request = retry(timeout=1, retry_delay=1, backoff=1)
     retry_request(self.connection.request)()
     """
-    def deco_retry(func):
+    if retry_exceptions is None:
+        retry_exceptions = RETRY_EXCEPTIONS
+    if retry_delay is None:
+        retry_delay = DEFAULT_DELAY
+    if timeout is None:
+        timeout = DEFAULT_TIMEOUT
+    if backoff is None:
+        backoff = DEFAULT_BACKOFF
+
+    timeout = max(timeout, 0)
+
+    def transform_ssl_error(func, *args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except ssl.SSLError:
+            exc = sys.exc_info()[1]
+
+            if TRANSIENT_SSL_ERROR in str(exc):
+                raise TransientSSLError(*exc.args)
```

Comment:
Shouldn't this have a fallback to raise the original exception?  Dunno how I missed that before.